engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/ProcessInstanceRestServiceImpl.java

```diff
@@ -275,6 +277,50 @@ public BatchDto setVariablesAsync(SetVariablesAsyncDto setVariablesAsyncDto) {
     return BatchDto.fromBatch(batch);
   }
 
+  @Override
+  public BatchDto correlateMessageAsync(CorrelationMessageAsyncDto correlationMessageAsyncDto) {
+    Map<String, VariableValueDto> variables = correlationMessageAsyncDto.getVariables();
+
+    VariableMap variableMap = null;
+    try {
+      variableMap = VariableValueDto.toMap(variables, processEngine, objectMapper);
+    } catch (RestException e) {
+      String errorMessage = String.format("Cannot set variables: %s", e.getMessage());
+      throw new InvalidRequestException(e.getStatus(), e, errorMessage);
+    }
+
+    String messageName = correlationMessageAsyncDto.getMessageName();
+    List<String> ids = correlationMessageAsyncDto.getProcessInstanceIds();
+    ProcessInstanceQuery runtimeQuery = toQuery(correlationMessageAsyncDto.getProcessInstanceQuery());
+    HistoricProcessInstanceQuery historyQuery = toQuery(correlationMessageAsyncDto.getHistoricProcessInstanceQuery());
+
+    RuntimeService runtimeService = getProcessEngine().getRuntimeService();
+
+    Batch batch = null;
+    try {
+      MessageCorrelationAsyncBuilder messageCorrelationBuilder = runtimeService
+        .createMessageCorrelationAsync(messageName)
+        .setVariables(variableMap);
+      if (ids != null) {
+        messageCorrelationBuilder.processInstanceIds(ids);
+      }
+      if (runtimeQuery != null) {
+        messageCorrelationBuilder.processInstanceQuery(runtimeQuery);
+      }
+      if (historyQuery != null) {
+        messageCorrelationBuilder.historicProcessInstanceQuery(historyQuery);
+      }
+      batch = messageCorrelationBuilder.correlateAllAsync();
+    } catch (BadUserRequestException | AuthorizationException e) {
+      throw e;
+    } catch (ProcessEngineException e) {
```

Comment:
```suggestion
    } catch (NullValueException e) {
```
`ProcessEngineException`s can happen in other situations than `null` values given. For instance, a persistence exception is probably not due to an invalid request. Should we throw all other `ProcessEngineException`s as is?